rust/gitxetcore/src/config/permission.rs

```diff
@@ -37,30 +54,125 @@ impl Permission {
         }
     }
 
-    pub fn check_path(&self, path: &Path) {
```

Comment:
Idea for creating $HOME/.xet/ -- if we are running as root (it would be under sudo, so HOME wouldn't be changed), then we should just make the .xet/ folder match the owner and permissions from the $HOME directory.

Reply:
This behavior is not the same between Linux and Mac.

Reply:
On Mac $HOME doesn't change. On Linux it seems to vary between distributions.

Reply:
This doesn't solve the problem that if a `.xet` is already owned by root. Also impersonating feels dangerous because... what if `.xet` as root is expected..

Reply:
If we are running as sudo root (so it's not in root's home directory, but it's in the user's home directory), we create the .xet/ folder to inherit the permissions of the home directory.   If it's in ~root/.xet, it then creates it to be owned by root. So intentionally running as root is fine.  Plus, if you're root, then it won't matter. 

The only scenario we are worried about here is when <user> runs as sudo, causing ~user/.xet/ to be owned by root.  The scenarios where $HOME is different because it's running under sudo are irrelevant, as the regular mode xet process would only access ~user/.xet/, and these would be different. 

If we do this, then it means we just have to worry about the current cases where this has already happened, as it won't be a problem again.  I'd be fine erroring out and asking the user to run chown on that directory in that case.

Reply:
Reworked to implement this.